brainreg_napari/register.py

```diff
@@ -409,14 +448,77 @@ def run(
             smoothing_sigma_floating,
             histogram_n_bins_floating,
             histogram_n_bins_reference,
+            save_original_orientation,
+            brain_geometry,
         )
         worker.returned.connect(add_image_layers)
         worker.start()
 
     @widget.reset_button.changed.connect
     def restore_defaults(event=None):
         for name, value in DEFAULT_PARAMETERS.items():
-            if name != "atlas_key":
+            if name not in ["atlas_key", "brain_geometry"]:
                 getattr(widget, name).value = value
 
+    @widget.check_orientation_button.changed.connect
+    def check_orientation(event=None):
+        """
+        Function used to check that the input orientation is correct. To do so it transforms the input data
+        into the requested atlas orientation, compute the average projection and displays it alongside the atlas. It
+        is then super easy for the super to identify which dimension should be swapped and avoid running the pipeline
+        on wrongly aligned data.
+        """
+
+        # Get viewer object
+        viewer = getattr(widget, "viewer").value
+        brain_geometry = getattr(widget, "brain_geometry").value
+
+        # Remove previous average projection layer if needed
+        ind_pop = []
+        for i, layer in enumerate(viewer.layers):
+            if layer.name in ['Ref. proj. 0', 'Ref. proj. 1', 'Ref. proj. 2',
+                              'Input proj. 0', 'Input proj. 1', 'Input proj. 2']:
+                ind_pop.append(i)
+            else:
+                layer.visible = False
+        for index in ind_pop[::-1]:
+            del viewer.layers[index]
+
+        # Load atlas and gather data
+        atlas = BrainGlobeAtlas('allen_mouse_25um')
+        if brain_geometry.value == "hemisphere_l":
+            atlas.reference[atlas.hemispheres == atlas.left_hemisphere_value] = 0
+        elif brain_geometry.value == "hemisphere_r":
+            atlas.reference[atlas.hemispheres == atlas.right_hemisphere_value] = 0
+        input_orientation = getattr(widget, "data_orientation").value
+        data = getattr(widget, "img_layer").value.data
```

Comment:
Same comments about `getattr`, `value` and `data` here. Might be because you're using `Enum`? But I would then question what the benefit is

Reply:
The Enum is enforced by Magic GUI.